packages/remix-express/server.ts

```diff
@@ -87,7 +87,11 @@ export function createRemixRequest(
   req: express.Request,
   res: express.Response
 ): Request {
-  let url = new URL(`${req.protocol}://${req.get("host")}${req.url}`);
+  let [, port] = req.get("host")?.split(":") ?? [];
```

Comment:
Should this try to get it directly from `X-Forwarded-Host`?  It looks like `req.hostname` strips it off, but wouldn't be run into issues if the proxy changed the port as well?

Locally I was testing this with the following `nginx` config (along with `127.0.0.1 example.com` in `/etc/hosts`):

```
http {
    server {
        listen 8080;
        server_name example.com;

        location / {
            proxy_pass http://localhost:3000;
            proxy_set_header X-Forwarded-Host example.com:8080;
        }
    }
}
```

When doing it that way it ends up with `http://example.com:3000` which works in this small setup since port 3000 is also exposed and `example.com` is an alias for `localhost` - but we can get the port if it exists using something like:

```js
  // req.hostname doesn't include port information so grab that from 
  // `X-Forwarded-Host` or `Host`
  let [,hostnamePort] = req.get('X-Forwarded-Host')?.split(':') ?? [];
  let [,hostPort] = req.get('host')?.split(':') ?? [];
  let port = hostnamePort || hostPort;
  // Use req.hostname here as it respects the "trust proxy" setting
  let resolvedHost = `${req.hostname}${port ? `:${port}` : ""}`;
  let url = new URL(`${req.protocol}://${resolvedHost}${req.url}`);
```

Thoughts?

Reply:
Yeah that seems like a reasonable approach :+1: